software/rocketlogger/rl_file.c

```diff
@@ -419,7 +419,7 @@ int rl_file_add_data_block(FILE *data_file, int32_t const *analog_buffer,
                 // on last sample of the window: average analog data and store
                 if ((i + 1) % aggregate_count == 0) {
                     for (int j = 0; j < RL_CHANNEL_COUNT; j++) {
-                        aggregate_analog[j] =
+                        ((int32_t*)aggregate_analog)[j] =
```

Comment:
I consider this partial in-place reuse of an int64_t buffer as storage for the the narrower int32_t results not easy to understand. Further, this is very fragile and error prone with respect to future changes.
* I suggest to introduce a dedicated an uint64_t buffer for the sample summation instead of reusing the same buffer for different data types.